Trophy pop-up and viewer enhancements

rainmakerv3 opened the pull request:
Partially addresses https://github.com/shadps4-emu/shadPS4/issues/2461

Trophy pop-up
1) show icon for trophy rarity next to the trophy art
2) center all elements vertically
3) minor spacing adjustments

Trophy viewer
1) show icon for trophy rarity instead of text
2) remove the hardcoded newlines that were being imported into the viewer from the XML
3) align left instead of center for trophy descriptions

Images, though patterned after the PS4 trophy icons, are originally modelled by discord user Tlarok. They are packaged inside the AppImage for Linux, inside the app bundle for Mac, and in a separate Resources folder which is packaged in the other builds, and copied by CMake command in the binary folder for locally compiled builds

sample photos:

Trophy pop-up (main):
![plat main](https://github.com/user-attachments/assets/96fa9eb8-0f73-4f97-91b0-6888f87dba4e)

Trophy pop-up (PR):
![plat PR](https://github.com/user-attachments/assets/6bd9163e-8544-425b-8459-d06822c97dfc)

Trophy viewer (main):
![viewer main](https://github.com/user-attachments/assets/7644b237-321d-4fdf-8df8-259c9f805b52)

Trophy viewer (PR):
![viewer PR](https://github.com/user-attachments/assets/fa5a35ce-42d4-4bd4-9505-ff8562668899)

rainmakerv3 committed:
Include trophy rarity icons in pop up, remove newlines from viewer

Fix layout

Update platinum.png

Fix linux and apple

rainmakerv3 committed:
Smaller type icons, center text vertically

rainmakerv3 committed:
use original icons

rainmakerv3 committed:
MacOS fixes

squidbus commented on .github/workflows/build.yml:
```diff
@@ -94,11 +94,14 @@ jobs:
       run: cmake --build ${{github.workspace}}/build --config ${{env.BUILD_TYPE}} --parallel $env:NUMBER_OF_PROCESSORS
 
+    - name: Move Files and Resources
+      run: "mkdir upload\n\t move build/Resources upload\nmove build/shadPS4.exe upload\n"
```
You can do multiline commands with `|`, just look for anywhere in the file with `run: |` for an example.

squidbus commented on CMakeLists.txt:
```diff
@@ -1144,3 +1148,19 @@ if (ENABLE_QT_GUI AND CMAKE_SYSTEM_NAME STREQUAL "Linux")
+add_custom_command(TARGET shadps4 POST_BUILD
+  COMMAND ${CMAKE_COMMAND} -E copy_directory
+    "${CMAKE_SOURCE_DIR}/Resources"
+    "$<TARGET_FILE_DIR:shadps4>/../Resources"
```
Typically files are added to the bundle resources like so:
```
    set_source_files_properties("your_source_file_path_here" PROPERTIES
        MACOSX_PACKAGE_LOCATION Resources/your_path_to_put_in_resources_here)
```

rainmakerv3 committed:
Address Review comments

Update build.yml

Update build.yml

Update build.yml

Update build.yml

Update build.yml

Update build.yml

Update build.yml

test

rainmakerv3 committed:
Move trophy type to leftmost and trophy art to rightmost

rainmakerv3 committed:
Embed resources

rainmakerv3 committed:
Revert packaging of resources with builds
